test/unit_test/common/test_retry.py

```diff
@@ -9,6 +9,9 @@
 # pylint: disable=global-variable-undefined, used-before-assignment
 
 
+num_tries = 0
```

Comment:
Since we are using num_tries to keep track of the number of calls, I found an alternative way of doing it

```python
def fail_with_key_error(lst):
    if len(lst) == 0:
        lst.append("dummy")
        raise KeyError("This is a key error")
    return True


@pytest.mark.parametrize("exceptions", [KeyError, (KeyError, ValueError), Exception])
def test_success(exceptions):
    input_lst = []
    assert retry_func(fail_with_key_error, args=[input_lst], max_tries=2, delay=1, exceptions=exceptions)
    assert len(input_lst) == 1
```

Reply:
I think it is ok to keep the global variable here for test only purpose.

Reply:
Personally, I like the global variable approach since it is simpler.

Reply:
Sure! I wasn’t sure if it would pass both the lint and codeql checks. But looks like it does https://github.com/microsoft/Olive/security/code-scanning?query=is%3Aopen+pr%3A1128

I agree, the global variable is better since it’s direct while the one I shared isn’t.